src/Cardano/Wallet/Binary.hs

```diff
@@ -496,6 +500,23 @@ encodeAddressPayload payload = mempty
     <> CBOR.encodeBytes payload
     <> CBOR.encodeWord32 (crc32 payload)
 
+
+encodeSignedTx :: Tx -> CBOR.Encoding
+encodeSignedTx tx = mempty
+    <> CBOR.encodeListLen 2
+    <> encodeTx tx
+    <> encodeList encodeTxWitness [tx]
+
+
+--data Witness = PubKeyWitness | ScriptWitness | RedeemWitness
+
+encodeTxWitness :: Tx -> CBOR.Encoding
+encodeTxWitness _tx = mempty
+    <> CBOR.encodeListLen 2
+    <> CBOR.encodeWord8 0 -- PK Witness
+    <> CBOR.encodeTag 0 -- what is this?
```

Comment:
A CBOR tag, sometimes there in some serialized structure of cardano-sl, value is always the same and hard-coded to 24. 

Don't ask why. I don't know :man_shrugging: 
Don't also ask why address payloads are serialized in CBOR, then hashed once, then hashed again with a different hashing algorithm, then re-serialized as part of the of the whole address and then, encoded to Base58 :no_mouth: :gun:

Reply:
lol 🙂